app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/registration/HowCanWeIdentifyYouController.scala

```diff
@@ -49,23 +49,30 @@ class HowCanWeIdentifyYouController @Inject() (
   def createForm(service: Service, journey: Journey.Value): Action[AnyContent] =
     authAction.ggAuthorisedUserWithEnrolmentsAction {
       implicit request => _: LoggedInUserWithEnrolments =>
-        Future.successful(Ok(howCanWeIdentifyYouView(ninoOrUtrForm, isInReviewMode = false, service, journey)))
+        populateView(service, journey, isInReviewMode = false)
     }
 
   def reviewForm(service: Service, journey: Journey.Value): Action[AnyContent] =
     authAction.ggAuthorisedUserWithEnrolmentsAction {
       implicit request => _: LoggedInUserWithEnrolments =>
-        subscriptionBusinessService.getCachedCustomsId.map { customsId =>
-          val ninoOrUtr = customsId match {
-            case Nino(id) => NinoOrUtr(Some(id), None, Some("nino"))
-            case Utr(id)  => NinoOrUtr(None, Some(id), Some("utr"))
-            case unexpected =>
-              throw new IllegalStateException(
-                "Expected a Nino or UTR from the cached customs Id but got: " + unexpected
-              )
-          }
-          Ok(howCanWeIdentifyYouView(ninoOrUtrForm.fill(ninoOrUtr), isInReviewMode = true, service, journey))
+        populateView(service, journey, isInReviewMode = true)
+    }
+
+  private def populateView(service: Service, journey: Journey.Value, isInReviewMode: Boolean)(implicit
+    hc: HeaderCarrier,
+    request: Request[_]
+  ) =
+    subscriptionBusinessService.getCachedCustomsId.map {
+      case Some(customsId) =>
+        val ninoOrUtr = customsId match {
+          case Nino(id) => NinoOrUtr(Some(id), None, Some("nino"))
+          case Utr(id)  => NinoOrUtr(None, Some(id), Some("utr"))
+          case unexpected =>
+            throw new IllegalStateException("Expected a Nino or UTR from the cached customs Id but got: " + unexpected)
```

Comment:
Has the customsId other types? Is this unexpected check necessary?

Nothing to do with this PR, it's old logic, just wondering in general

Reply:
Who knows?

Reply:
Yeah, it's possible, this trait is extended by more than UTR and NINO